internal/pkg/runtime/engine/fakeroot/engine_linux.go

```diff
@@ -294,10 +296,47 @@ func (e *EngineOperations) StartProcess(_ int) error {
 	if err != nil {
 		return fmt.Errorf("failed to mount %s to /root: %s", e.EngineConfig.Home, err)
 	}
+	// read the real binfmt_misc
+	entries, err := os.ReadDir(binfmtMisc)
+	if err != nil {
+		return err
+	}
+	content := map[string][]byte{}
+	for _, entry := range entries {
+		name := entry.Name()
+		if name == "register" {
+			continue
+		}
+		f := filepath.Join(binfmtMisc, name)
+		b, err := os.ReadFile(f)
+		if err != nil {
+			sylog.Warningf("%v", err)
+			continue
+		}
+		content[entry.Name()] = b
+	}
 	err = syscall.Mount("proc", "/proc", "proc", syscall.MS_NOSUID|syscall.MS_NOEXEC|syscall.MS_NODEV, "")
 	if err != nil {
 		return fmt.Errorf("failed to mount proc filesystem: %s", err)
 	}
+	// save a copy of binfmt_misc
+	binfmtMisc = "/tmp/sys/fs/binfmt_misc"
```

Comment:
Using a path that's shared by all processes on a machine is not a good idea.  They will clash.

However I'm not sure what else to suggest as a good solution to this problem.  @cclerget can you?

Reply:
The usual path is _also_ shared by all processes (being a kernel one), but there is a race during creation now.

i.e. Previously it was leaving the old settings, so I now delete any old directory - that probably needs a flock

Reply:
The usual path is a special kernel-created path that is different for every process, like a mount namespace.  Apptainer uses a mount namespace too including a tmpfs filesystem but I'm not sure if it is created at that point.

Reply:
It needs to be mounted, but unfortunately binft_misc is not yet part of user namespaces so it is not allowed as rootless (like /proc)

Reply:
But that's ok, since we don't need to register anything from within the container. A read-only copy of what is already loaded in the kernel is fine...

Reply:
@afbjorklund why not just inject a bind mount for `/proc/sys/fs/binfmt_misc` on command line when using `--fakeroot`  instead of duplicating the directory content here?

Reply:
@cclerget Wouldn't that be a problem since `/proc` is a new mount?  A bind at this point would just come from the new `/proc`, and a bind before would be covered up by the new `/proc`.  Maybe it could be bound to a temporary place before and bound again from there to the original place afterward.  But where's a good temporary place?

Reply:
The new /proc has an empty directory for the binfmt_misc, which is what lead to the "status" not found error. The rootless user cannot mount the real binfmt_misc, since it is not covered by the namespace. And trying to populate the directory directly leads to errors, since it is not a normal directory (you go through "register")

Reply:
I did not try a temporary mountpoint.

Reply:
doing the equivalent of this in Go should do it:

```
tmpdir=$(mktemp -d bind-mount-XXXX)
mount -t proc proc $tmpdir
mount --bind /proc/sys/fs/binfmt_misc/ ${tmpdir}/sys/fs/binfmt_misc
mount --move ${tmpdir} /proc
rm -rf $tmpdir
```

Reply:
Took a while, but doing the implementation as suggested (except rmdir, not rm -rf). Seems to work fine!